backpack.tf extended sorting.user.js

```diff
@@ -1268,4 +1268,246 @@ class</a></li>
+        let itemId = currentItem.getItemId();
+        let killstreakTier = currentItem.getKillstreakTier();
+        let currentPath = window.location.pathname;
+
+        $(".stats-header-controls").append(`
+     <br>
+     <div class="btn-group btn-group-sm stats-killstreak-list" style="margin: 0 0 12px"></div>
+     `);
+
+        switch(currentItem.getItemSlot()){
+
+            case 0:
+                break;
+
+            case -1: {
+
+                let weaponName = trimSpaces(document.querySelector(".stats-breadcrumbs").children[1].innerHTML);
+                let weaponDefIndex = window.location.pathname.match(/\d+$/g);
+
+                $(".stats-killstreak-list").append(`
+          <a class="btn btn-variety q-440-text-4" href="https://backpack.tf/stats/Unique/Killstreak%20Kit/Tradable/Non-Craftable/1-${weaponDefIndex}"> Basic </a>
+     <a class="btn btn-variety q-440-text-7" href="https://backpack.tf/stats/Unique/Killstreak%20Kit/Tradable/Non-Craftable/2-${weaponDefIndex}"> Specialized </a>
+     <a class="btn btn-variety q-440-text-11" href="https://backpack.tf/stats/Unique/Killstreak%20Kit/Tradable/Non-Craftable/3-${weaponDefIndex}"> Professional </a>
+     `);
+                document.querySelector(".stats-killstreak-list").children[killstreakTier-1].classList.add("active");
+                if(killstreakTier > 1){
+                    $(".stats-header-controls").append(`
+     <a class="btn btn-default" href="https://backpack.tf/stats/Unique/Fabricator/Tradable/Craftable/${6520 + 3 * (killstreakTier - 1)}-6-${weaponDefIndex}" style="margin: 0px 0px 12px;">
+     <i class="fa fa-bar-chart"></i>
+     Fabricator Stats
+     </div>
+     `);
+                }
+
+                $(".stats-header-controls").append(`
+     <br>
+     <div class="temp-weaponlist-div" style="margin: 0 0 12px">Loading applicable weapons...</div>
+     `);
+                $(".temp-weaponlist-div").load(`https://backpack.tf/overview/${weaponName} .overview-quality-list`,function(){
+                    for(let button of this.firstChild.children){
+                        button.href = appendKSToPath(killstreakTier, button.getAttribute("href"), 0);
+                    }
+                    this.firstChild.classList.remove("overview-quality-list");
+                    this.firstChild.classList.remove("expanded");
+                    $(this.firstChild).append(`<a class="btn btn-variety" id="btn-expand-list">
+                                          <i class="fa fa-ellipsis-h"></i>
+                                          </a>`);
+                    this.outerHTML = this.innerHTML;
+                    $("#page-content").append(`<script>
+    $('#btn-expand-list').click(function () {
+        var $parent = $(this).parent();
+
+        if ($parent.hasClass('expanded')) {
+            $parent.removeClass('expanded');
+        } else {
+            $parent.addClass('expanded');
+        }
+    });
+</script>`);
```

Comment:
You could shorten this to the following:

```suggestion
                for(let button of this.firstChild.children){
                    button.href = appendKSToPath(killstreakTier, button.getAttribute("href"), 0);
                }
		$(this.firstChild).removeClass("overview-quality-list expanded")
			.append(`<a class="btn btn-variety" id="btn-expand-list"><i class="fa fa-ellipsis-h"></i></a>`);

		  $('#btn-expand-list').click(function () {
			  $(this).parent().toggleClass("expanded");
		  });
```

(Use JQuery to remove the classes and chain the methods, additionally use an JQuery Event to toggle the class, instead of adding an additional script tag to remove and add classes by hand) 

If I am not wrong removing the outer container class should be redundant (style wise) so I removed it, but you can also keep it.